suite-common/wallet-utils/src/accountUtils.ts

```diff
@@ -1153,3 +1155,59 @@ export const isTokenMatchesSearch = (token: TokenInfo, search: string) => {
         token.policyId?.toLowerCase().includes(search)
     );
 };
+
+export const isDebugOnlyAccountType = (
+    accountType: AccountType,
+    symbol?: NetworkSymbol,
+): boolean => {
+    if (!symbol) return false;
+
+    const network = (networks as Networks)?.[symbol];
+
+    if (!network) return false;
+
+    const accountTypeInfo = network.accountTypes[accountType];
+
+    return !!accountTypeInfo?.isDebugOnlyAccountType;
+};
+
+type FilterRecieveAccountsProps = {
+    accounts: Account[];
+    deviceState: string | undefined;
+    receiveNetwork?: string;
+    isDebug: boolean;
+    receiveNetworks: Network[];
+};
+
+export const filterReceiveAccounts = ({
+    accounts,
+    deviceState,
+    receiveNetwork,
+    isDebug,
+    receiveNetworks,
+}: FilterRecieveAccountsProps): Account[] => {
+    const isSameDevice = (account: Account) => account.deviceState === deviceState;
+    const isSameNetwork = (account: Account) => account.symbol === receiveNetwork;
+    const isDebugAndIsAccountDebugOnly = (account: Account) =>
+        isDebugOnlyAccountType(account.accountType, account.symbol) && isDebug;
+    const isNotDebugOnlyAccount = (account: Account) =>
+        !isDebugOnlyAccountType(account.accountType, account.symbol);
```

Comment:
This is quite confusing to read, it all sounds the same :sweat_smile:
I know you only moved the code around, but it's an opportunity to simplify: both of these fns can be replaced with one, which will do:
`isDebug ||  !isDebugOnlyAccountType(account.accountType, account.symbol);`

Reply:
Thank you, it's a nice note